pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -444,9 +448,34 @@ func getFirstContainerWithSourceVolume(containers []corev1.Container) (string, s
 
 // Delete deletes the component
 func (a Adapter) Delete(labels map[string]string) error {
-	if !utils.ComponentExists(a.Client, a.ComponentName) {
-		return errors.Errorf("the component %s doesn't exist on the cluster", a.ComponentName)
+	spinner := log.Spinner(fmt.Sprintf("Deleting devfile component %s", a.ComponentName))
+	defer spinner.End(false)
+
+	errorMsg := fmt.Sprintf("Component %s does not exist", a.ComponentName)
+
+	componentExists, err := utils.ComponentExists(a.Client, a.ComponentName)
+	if kerrors.IsForbidden(err) {
+		klog.V(4).Infof("Resource for %s forbidden", a.ComponentName)
+		// log the error if it failed to determine if the component exists due to insufficient RBACs
+		log.Infof("%s: %v", errorMsg, err)
+		spinner.End(true)
```

Comment:
since this is an error, shouldn't this be false?

Reply:
I had it as `true` because `Forbidden` is a special case when no namespace exist which is similar to no component exist from delete's perspective and we dont want to show a cross check mark on `odo delete`

Reply:
@kadel do you have any recommendation on whether we should show a ✗ or ✓; when 
1. component doesnt exist
2. namespace doesn't exist

In both instances, we're logging information on the screen.

Reply:
I think namespace error can be considered a genuine error as this will inform the user that he is not logged in the cluster.

Reply:
I've updated the spinners and the `log.Infof` to `log.Warningf` since I chatted with Tomas and we wanted a spinner with warning symbol. But looks like odo spinner either does a ✗ or ✓, not ⚠. And looks like the spinner.WarningStatus is broken:
```
✗  Deleting devfile component testm [2ms] [this is a warning spinner example]
```

it doesn't print `[this is a warning spinner example]` in yellow or have a ⚠ suffix :( we need to fix this on master

So I went with :
```
Maysuns-MacBook-Pro:springboot-ex maysun$ odo delete --project ga -af 
 ✗  Deleting devfile component testm [1ms]
 ⚠  Component testm does not exist: deployments.apps "testm" is forbidden: User "developer" cannot get resource "deployments" in API group "apps" in the namespace "ga"
 ✓  Successfully deleted env file
 ✓  Successfully deleted devfile.yaml file

Maysuns-MacBook-Pro:springboot-ex maysun$ odo delete -af
 ✗  Deleting devfile component testm [3ms]
 ⚠  Component testm does not exist
 ✓  Successfully deleted env file
 ✓  Successfully deleted devfile.yaml file
```

@kadel @mik-dass lmk, if this looks alright!

Reply:
> So the current PR does this (end spinner as false and print a warning msg):
> 
> ```
> Maysuns-MacBook-Pro:springboot-ex maysun$ odo delete --project ga -af 
>  ✗  Deleting devfile component testm [1ms]
>  ⚠  Component testm does not exist: deployments.apps "testm" is forbidden: User "developer" cannot get resource "deployments" in API group "apps" in the namespace "ga"
>  ✓  Successfully deleted env file
>  ✓  Successfully deleted devfile.yaml file
> 
> Maysuns-MacBook-Pro:springboot-ex maysun$ odo delete -af
>  ✗  Deleting devfile component testm [3ms]
>  ⚠  Component testm does not exist
>  ✓  Successfully deleted env file
>  ✓  Successfully deleted devfile.yaml file
> ```

this looks good

/approce

Reply:
/approve

Reply:
> Maysuns-MacBook-Pro:springboot-ex maysun$ odo delete --project ga -af 
 ✗  Deleting devfile component testm [1ms]
 ⚠  Component testm does not exist: deployments.apps "testm" is forbidden: User "developer" cannot get resource "deployments" in API group "apps" in the namespace "ga"
 ✓  Successfully deleted env file
 ✓  Successfully deleted devfile.yaml file

Maysuns-MacBook-Pro:springboot-ex maysun$ odo delete -af
 ✗  Deleting devfile component testm [3ms]
 ⚠  Component testm does not exist
 ✓  Successfully deleted env file
 ✓  Successfully deleted devfile.yaml file

This looks good :+1:

Reply:
Thx, will keep it as is 👍